src/foam/nanos/auth/UserAndGroupAuthService.js

```diff
@@ -258,9 +248,33 @@ foam.CLASS({
     {
       name: 'validatePassword',
       javaCode: `
-        if ( SafetyUtil.isEmpty(potentialPassword) || ! (Pattern.compile(PASSWORD_VALIDATE_REGEX)).matcher(potentialPassword).matches() ) {
-          throw new RuntimeException(PASSWORD_VALIDATION_ERROR_MESSAGE);
+        PasswordPolicy passwordPolicy = null;
+
+        // Retrieve the password policy from the user and group when available
+        if ( user != null ) {
+          Group ancestor = user.findGroup(x);
+          if ( ancestor != null ) {
```

Comment:
When can this happen? Are users allowed to not have groups, or is that an error in the system? If the latter then IMO this should be a check inside `findGroup` itself